src/librustc_mir/transform/inline.rs

```diff
@@ -546,22 +547,31 @@ impl<'a, 'tcx> Inliner<'a, 'tcx> {
         Operand::Consume(cast_tmp)
     }
 
-    fn make_call_args(&self, args: Vec<Operand<'tcx>>,
-                      callsite: &CallSite<'tcx>, caller_mir: &mut Mir<'tcx>) -> Vec<Operand<'tcx>> {
-        let tcx = self.tcx;
+    fn make_call_args(
+        &self,
+        args: Vec<Operand<'tcx>>,
+        callsite: &CallSite<'tcx>,
+        caller_mir: &mut Mir<'tcx>,
+    ) -> Vec<Operand<'tcx>> {
         // FIXME: Analysis of the usage of the arguments to avoid
         // unnecessary temporaries.
-        args.into_iter().map(|a| {
-            if let Operand::Consume(Lvalue::Local(local)) = a {
+
+        fn create_temp_if_necessary<'a, 'tcx: 'a>(
```

Comment:
Nit: Could we move this to a method (and then get rid of the `tcx` argument)? (I just find it harder to read things "nested" funtions, that's all.)

Reply:
Also, good opportunity to add a comment (the surrounding code is a bit...comment-poor). Perhaps something like:

```
/// If `arg` is already a temporary, returns it. Otherwise, introduces a fresh
/// temporary `T` and an instruction `T = arg`, and returns `T`.
```

(Does that sound accurate?)

Reply:
Finally, can we change the return value to `Lvalue`, to reflect the fact that this *always* returns an `Operand::Consume`?